lib/request-state.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+var Map = require('immutable').Map;
+var History = require('./history');
+
+/**
+ * A Request State is an immutable state corresponding to a request processing or processed by the system.
+ *
+ * @param {string} status - The current status of the request
+ * @param {Map.<string, Job>} jobs - A map of all the jobs related to this request
+ * @param {History} history - A human-readable history for this request
+ * @param {Map.<string, *>} metadata - The metadata extracted by Specberus for this request
+ */
+function RequestState(status, jobs, history, metadata) {
+  if (typeof this !== 'object') {
+    throw new TypeError('RequestState must be constructed via new');
+  }
+
+  this.status = typeof status !== 'undefined' ? status : '';
```

Comment:
I forgot to add: if someone overwrites `undefined` on you, you actually want it to create a bug. Because that will give you a reason to then go break their fingers off with a sledgehammer, as they so totally deserve.

In general for these things I follow the simple rules in those two sections: http://contribute.jquery.org/style-guide/js/#equality